macro/vignettes/pfSI.Rmd

```diff
@@ -0,0 +1,385 @@
+---
+title: "Human Module with SI Dynamics"
+output:
+  html_notebook:
+    df_print: paged
+  pdf_document:
+    df_print: kable
+    number_sections: yes
+header-includes:
+  - \usepackage{palatino}
+bibliography: pfSI.bib
+---
+```{r load_libraries, include = FALSE}
+library(data.table)
+```
+
+# Overview
+
+This notebook describes a MASH module for humans who can be
+either susceptible or infectious. We call it `pfSI`.
+
+There is an existing module in a former implementation of MASH,
+so this document will walk through that implementation in order
+to write out this one.
+
+We ask here the questions we ask of any model, be it discrete, continuous,
+deterministic or stochastic. What are the states,
+what are the transitions, and how do you initialize the model?
+The implementation given is continuous time, so it creates individual events,
+which should make it easier to track what's happening.
+
+
+# Problem Description
+
+## State Space
+
+The first thing to define for this model is the set of possible
+states a human can be in. We split these into three categories.
+Some properties of a person are unchanging, and we call them
+_frailty._ Some properties of a person can be figured out. For instance,
+if biting weight is a function of age, then you can figure it out from
+the age without storing biting weight. We call these _predictable variables._
+The rest is state.
+
+* Frailty
+  + birth date
+  + mosquito-to-human transmission efficiency, _b_
+  + human-to-mosquito transmission efficiency, _c_
+* State
+  + alive or dead - a boolean.
+  + disease status - as susceptible, latent, infectious, or protected from infection
+  + fever - reachable from the infectious state (or latent?)
+  + PE vaccination - lowers biting weight by factor (1 - pe_block_pf)
+* Predictable Variables
+  + biting weight - this represents the amount of skin this person has,
+    so it changes with age. Lowered by PE vaccination
+  + kappa - if the person is infected and _not in a resevoir,_ this is c x (biting weight).
+    Lowered by GS vaccination by a factor c = c x (1 - gs_block_pf).
+  + EIR - the rate the person is getting bitten. For someone in a resevoir,
+    this is the resevoir EIR. For someone not in a resevoir, this
+    is max(0, beta x biting_weight / patch_biting_weight).
+  + incidence - the number of infectious people in a patch. This is the sum of all infectious.
+
+
+## Event Space
+
+Here, variable names are sometimes shortened. Mean is mn, and variance is vr.
+There is a notation for events that says what state you leave and what
+you enter. The minus is leave and the + is enter.
+
+* Infection
+  + infection - from state L to I (-L, +I), exponential with rate duration
+  + clearance - from state I to S (-I, +S), exponential with rate duration_pf
+  + fever - (-I, +F, +I), exponential with rate mnFever, which is incomplete by U(0, 1) < fever_pf.
```

Comment:
Might suggest saying: "fever - (-I, +F, +I), exponential with rate mnFever, if U(0,1) < fever_pf; another way of saying this is that time to fever follows an exponential distribution with parameter `mnFever` but whose CDF F(infinity) = fever_pf."